Added the files

HarshaSomaraju opened the pull request:
1. Database.py which connects to the MYSQL server and creates the database.
2. intern.py which has the code to use the api to get the data and add to the database.
3. crontab which has the code to run the script for every 5 minutes
4. README containing information about all files and how to execute.

HarshaSomaraju committed:
Added the files
1. Database.py which connects to the MYSQL server and creates the database.
2. intern.py which has the code to use the api to get the data and add to the database.
3. crontab which has the code to run the script for every 5 minutes
4. README containing information about all files and how to execute.

dwividha commented on Python_DataEngineering/Database.py:
```diff
@@ -0,0 +1,77 @@
+    "CREATE TABLE `bikes_availability` ("
+    "  `bikeid` int(11) NOT NULL,"
+    "  `nam` varchar(50) NOT NULL,"
+    "  `lat` varchar(50) NOT NULL,"
```
Why not store this as float or double?

HarshaSomaraju replied:
Yes, i changed it to demical type.

dwividha commented on Python_DataEngineering/Database.py:
```diff
@@ -0,0 +1,77 @@
+    "   `TimeSnap` datetime,"
+    "   `nbBikes` int(3),"
+    "   `nbEmptyDocks`  int(3),"
+    "  PRIMARY KEY (`TimeSnap`)"
```
This might create problems - what if 2 bike stands are being updated at the same time? If your primary key is only timestamp, you can only store 1 of them

HarshaSomaraju replied:
Yes, that is true. I removed the primary key as TimeSnap and added a unique constraint of BikeId and TimeSnap. Different combination of bikeId and TimeSnap will create new rows. It generally means new rows are not added until and unless it is either a new bikeId or an old bikeId with latest data.

dwividha commented on Python_DataEngineering/Database.py:
```diff
@@ -0,0 +1,77 @@
+    ")")
+
+
+cnx = mysql.connector.connect(user='root',password='1234')
```
DB Secrets should be stored in environment variables for security, rather than in code

HarshaSomaraju replied:
I added the variables in environment and used those variables. Thanks for it.

dwividha commented on Python_DataEngineering/intern.py:
```diff
@@ -0,0 +1,47 @@
+    id = int(bd['id'].split('_')[-1])
+    name = str(bd['commonName'])
+    aP = bd['additionalProperties']
+    installed = aP[1]['value']
```
You're assuming that the format in which the bike data comes will be constant (1 for installed, 2 for locked, etc), what if this changes? Can you make it more generic so that even if the order is changed, your code will still hold good?

HarshaSomaraju replied:
That will be problem. I changed the code so that it will work even if the order is changed.

HarshaSomaraju committed:
1. Changed 'lat' and 'longitude' to decimal type.
2. Removed the primary key as TimeSnap and added a unique constraint of BikeId and TimeSnap. Different combination of bikeId and TimeSnap will create new rows. It generally means new rows are not added until and unless it is either a new bikeId or an old bikeId with latest data.
3. Used environment variables for username and password for database.
4. Changed the code in intern.py to be more generic and work even if the order is changed from the data retrieved using api.

HarshaSomaraju committed:
Changed the lat and longitude to decimal type in table bikes_availability
